### ISSUE TITLE
@graph decorator error message incorrect

### ISSUE THREAD

**Issue by will-regal-voice** [dagster-io/dagster#18711]
### Dagster version

dagster, version 1.4.17

### What's the issue?

When I use the @graph decorator with a Nothing In, I get the following error message:
```
dagster._core.errors.DagsterInvalidDefinitionError: @graph 'downstream_graph' decorated function does not have argument(s) 'start'. @graph-decorated functions should have a keyword argument for each of their Ins, except for Ins that have the Nothing dagster_type. Alternatively, they can accept **kwargs.
```

### What did you expect to happen?

Not get the error message, because the In I am passing has the Nothing dagster_type

### How to reproduce?

```
@op()
def upstream_op():
    pass

@op()
def downstream_op_1():
    pass

@op()
def downstream_op_2():
    pass

@graph(ins={"start": In(Nothing)})
def downstream_graph():
    downstream_op_1()
    downstream_op_2()

@job()
def full_job():
    downstream_graph(start=upstream_op())
```

To avoid it, I can pass it through like so (but this is inconsistent with the error message):
```
@op()
def upstream_op():
    pass

@op(ins={"start": In(Nothing)})
def downstream_op_1():
    pass

@op(ins={"start": In(Nothing)})
def downstream_op_2():
    pass

@graph(ins={"start": In(Nothing)})
def downstream_graph(start):
    downstream_op_1(start)
    downstream_op_2(start)

@job()
def full_job():
    downstream_graph(start=upstream_op())

```

### Deployment type

Local

### Message from the maintainers

Impacted by this issue? Give it a 👍! We factor engagement into prioritization.

**Comment by sryza** [comment]
@alangenfeld  - do I recall correctly that you fixed something related to this error?

@will-regal-voice - it is expected that an error will show up in this case. Even though it has a `Nothing` type, it still needs to be wired up to one of the inputs of the ops inside the graph, in order for there to be a dependency.  Graphs themselves don't have dependencies that are independent of the dependencies of the ops inside of them. Before executing, Dagster needs to be able flatten out all the dependencies to the op level, in order to create a set of steps.

**Comment by alangenfeld** [comment]
https://github.com/dagster-io/dagster/commit/ee6f0278e5513af6f8b52ec713f996e617f31773
released in `1.5.5`

**Comment by will-regal-voice** [comment]
confirming the error message was fixed by upgrading - thanks. and also thanks the explanation @sryza
